### PR TITLE
Custom burgers - FoodSequence system

### PR DESCRIPTION
**Pull request by TheShuEd**
## About the PR
Added a new food crafting system that allows you to herd different types of food into a single entity. The system is versatile and can be used for kebabs, sandwiches or just dishes on a plate, among other things. But in this PR, only burgers are added.

## Why / Balance
Fuck microwave, its new era of procedural food

## Media
image

image

https://github.com/user-attachments/assets/406c65d2-76e0-4b38-a86d-6c68b8d209e0

## Breaking changes
`Trash` field in FoodComponent is now HashSet<>, so all food prototypes need changes from
```yml
- type: Food
  trash: TrashId
```
to
```yml
- type: Food
  trash:
  - TrashId
```

**Changelog**
:cl:
- add: You cat cut burger bun into two halfs, and make custom burgers now!

### CODE REVIEW

**Review by Emisse — APPROVED** [review]

**Comment by TheShuEd** [comment]
Edit: Ingredient names are no longer repeated.

**Comment by TheShuEd** [comment]
https://github.com/user-attachments/assets/a18b0a90-f264-4b3b-aace-26e03455595f

also added noRot: true and drawdepth: Mobs to burgers

File: Resources/Locale/en-US/nutrition/components/food-sequence.ftl [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+food-sequence-burger-content-raw-meat = raw
+food-sequence-burger-content-carp = carpo
+food-sequence-burger-content-bear = bear
+food-sequence-burger-content-penguin = peng
+food-sequence-burger-content-chiken = chiken
+food-sequence-burger-content-duck = duck
+food-sequence-burger-content-corgi = corgi
+food-sequence-burger-content-crab = crabs
+food-sequence-burger-content-goliath = goli
+food-sequence-burger-content-dragon = gragon
```
Comment (Hrosts) [inline Resources/Locale/en-US/nutrition/components/food-sequence.ftl:17]:
```suggestion
food-sequence-burger-content-dragon = dragon
```
Literally unplayable

**Comment by Moomoobeef** [comment]
this is awesome. I always intended to make a doc for a cooking system inspired by how [project zomboid does it](https://pzwiki.net/wiki/Cooking_(crafting)#Evolved_recipes) but I never got around to it. This is similar though (for burgers anyway) so if I ever do get around to it this might be a good foundation!

**Comment by Moomoobeef** [comment]
@TheShuEd I just had a good idea as well if you want to add this to this PR; as you stack a burger taller and taller the do-over should be longer, because it makes sense that stacking a burger twice as tall as oneself would be challenging.

**Pull request merged**